odo/backends/csv.py

```diff
@@ -366,8 +367,19 @@ def resource_glob(uri, **kwargs):
 @convert.register(chunks(pd.DataFrame), (chunks(CSV), chunks(Temp(CSV))),
                   cost=10.0)
 def convert_glob_of_csvs_to_chunks_of_dataframes(csvs, **kwargs):
-    data = [partial(convert, chunks(pd.DataFrame), csv, **kwargs) for csv in csvs]
-    return chunks(pd.DataFrame)(data)
+    f = partial(convert, chunks(pd.DataFrame), **kwargs)
+
+    def df_gen():
+        dsk = {}
+        p = []
+        for n, csv_ in enumerate(csvs):
```

Comment:
Ok. Then it might be good to add a comment that explains why the loop is necessary.